classes/fields/datetime.php

```diff
@@ -345,15 +338,9 @@ public function pre_save( $value, $id = null, $name = null, $options = null, $fi
 		// Value should always be passed as storage format since 2.7.15.
 		$format = static::$storage_format;
 
-		if ( ! empty( $value ) && ( 0 === (int) pods_v( static::$type . '_allow_empty', $options, 1 ) || ! in_array(
-			$value, array(
-				'0000-00-00',
-				'0000-00-00 00:00:00',
-				'00:00:00',
-			), true
-		) ) ) {
+		if ( ! $this->is_empty( $value ) ) {
 			$value = $this->convert_date( $value, static::$storage_format, $format );
-		} elseif ( 1 === (int) pods_v( static::$type . '_allow_empty', $options, 1 ) ) {
+		} elseif ( pods_v( static::$type . '_allow_empty', $options, 1 ) ) {
```

Comment:
The option could come as a string and evaluate as true though, right?

Reply:
Why would it come as a string? And even if it did come as `'0'` it would still be `false`.
I think it's good to simplify the way we validate boolean types.

Reply:
I double checked '0' evaluates to false, nevermind!